flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/hints/batch/JoinHintTestBase.java

```diff
@@ -232,6 +233,21 @@ public void testJoinHintWithMultiUnknownTableNamesMixedWithValidTableNames() {
         verifyRelPlanByCustom(String.format(sql, getTestSingleJoinHint()));
     }
 
+    @Test
+    public void testJoinHintWithConfigQueryHintsIgnore() {
+        util.tableEnv().getConfig().set(TableConfigOptions.TABLE_QUERY_HINTS_IGNORE, true);
+
+        thrown().expect(ValidationException.class);
+        thrown().expectMessage(
+                        String.format(
+                                "The '%s' hint is allowed only when the config option '%s' is set to true.",
```

Comment:
This I didn't get: above the config option is set to `true` and here we are expecting exception saying that the hint is allowed only when config option is set to `true`...
What should be done to have it working without exception then?